### DISCUSSION TITLE
Is it possible to validate all query parameters at once?

### DISCUSSION THREAD

**Discussion by flashgorman-chorus** [fastapi/fastapi#6019] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Union

from fastapi import FastAPI

app = FastAPI()


def my_validator(param1: int):
    raise HTTPException(status_code=400, detail="Bad value")


@app.get("/foo")
def read_item(param1: int = Depends(my_validator), param2: int):
    pass
```

### Description

When I use `Depends` for custom query parameter validation, only the first error is reported.  In the example above, if I try to call the API without any value for `param2`, that never gets reported due to the exception raised by my custom validator used for `param1`.

Is it possible to get FastAPI to validate all query parameters - even those that use custom validators via `Depends` - so I can communicate ALL of the problems back to the caller instead of one at a time?

### Operating System

macOS

### FastAPI Version

0.89.1

### Python Version

3.7.14

**Comment by SquarishRectangle** [comment]
I cannot reproduce this issue on my machine. It behaves correctly here. 
```
http://127.0.0.1:8000/foo?param1="abc"

returns:
{"detail":[{"loc":["query","param1"],"msg":"value is not a valid integer","type":"type_error.integer"},{"loc":["query","param2"],"msg":"field required","type":"value_error.missing"}]}
```

Operating System
Windows, Linux

Operating System Details
Ubuntu running on WSL

FastAPI Version
0.89.1

Python Version
3.10.9

**Reply by flashgorman-chorus** [reply]
My apologies.  I realize now this situation only exists when using custom query parameter validation using `Depends`.  I have updated the description accordingly.

What I observe is if you raise from a custom query parameter validator, the other (or remaining) validation is not performed.
